configdns-v1/zone.go

```diff
@@ -765,7 +765,11 @@ func (zone *Zone) removeTxtRecord(record *TxtRecord) error {
 }
 
 func (zone *Zone) PreMarshalJSON() error {
-	zone.Zone.Soa.Serial = int(time.Now().Unix())
+	if zone.Zone.Soa.Serial > 0 {
```

Comment:
Perhaps this check should actually be:

```go
currentTime := int(time.Now().Unix())
if zone.Zone.Soa.Serial == currentTime {
    zone.Zone.Soa.Serial = zone.Zone.Soa.Serial + 1
} else {
    zone.Zone.Soa.Serial = currentTime
}
```

Reply:
That won't work concurrently if two goroutines call this function during the same second, they will end up with the same serial number. So I think the original version where it always increments by one is the best option. 

Also after thinking about this more, I realised we actually have to restrict the concurrency of calls to `Zone.Save()` all together because the requests need to come in the same order, to ensure the serial is incremented in order. The original version of this PR will ensure a unique serial number, but it won't ensure they are submitted to the API in ascending order. So I've added a mutex to restrict calls to `Zone.Save()` so only one can run at a time.